### PR TITLE
refactor: extract a `run` fn to remove duplicates

### PR DESCRIPTION
**Pull request by jukanntenn**

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -3,3 +3,111 @@ pub mod inverse;
...
+    });
+    rx
+}
+
+fn future_ping() -> Receiver<String> {
+    let (tx, rx) = mpsc::channel::<String>();
+    thread::spawn(move || {
+        let s30 = Duration::from_secs(30);
+        loop {
+            tx.send("{\"op\":\"ping\"}".into()).unwrap();
```
Comment (EricCrosson) [inline src/lib.rs:108]: It's a little annoying that Bybit wants separate formats for spot and future pings :dizzy:

File: src/lib.rs [hunk 2]
```diff
@@ -3,3 +3,111 @@ pub mod inverse;
...
+        ws.write_message(Message::Text(subscription.clone()))
+            .unwrap();
+    }
+
+    loop {
+        if let Ok(ping) = rx.try_recv() {
+            ws.write_message(Message::Text(ping)).unwrap();
+        }
+
+        if let Ok(msg) = ws.read_message() {
```
Comment (EricCrosson) [inline src/lib.rs:70]: I've found that if the `read_message` takes long enough (10 minutes) to return the next websocket message, than we starve the sending of ping on line 67 and Bybit will terminate the websocket because it thinks the client is not responsive.

The fix seems out of scope for this refactoring PR

**Review by EricCrosson — APPROVED** [review]

File: src/lib.rs (hunk 2 above)
Comment (jukanntenn) [reply src/lib.rs:70]: Yeah, when testing, I found the WebSocket connection will be terminated after about 30s if ping message did not send. I am planing to fix it in next PR.
